osde2e/splunk_forwarder_operator_tests.go

```diff
@@ -0,0 +1,230 @@
+		//checkClusterServiceVersion(h, operatorNamespace, operatorName)
+
+		// TODO: post osde2e-common library add upgrade check
+		//checkUpgrade(helper.New(), "openshift-splunk-forwarder-operator",
+		//	"openshift-splunk-forwarder-operator", "splunk-forwarder-operator",
+		//	"splunk-forwarder-operator-catalog")
+	})
+
+	ginkgo.It("admin should be able to manage SplunkForwarders CR", func(ctx context.Context) {
+		sf := makeMinimalSplunkforwarder("SplunkForwarder", "splunkforwarder.managed.openshift.io/v1alpha1", testsplunkforwarder)
+		err := addSplunkforwarder(ctx, sf, "openshift-splunk-forwarder-operator")
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	ginkgo.It("test SplunkForwarders CR, if exists, should be deleted successfully", func(ctx context.Context) {
+		err := deleteSplunkforwarder(ctx, testsplunkforwarder, operatorNamespace)
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	ginkgo.It("dedicated admin should not be able to manage SplunkForwarders CR", func(ctx context.Context) {
+
+		sf := makeMinimalSplunkforwarder("SplunkForwarder", "splunkforwarder.managed.openshift.io/v1alpha1", dedicatedadminsplunkforwarder)
+		err := dedicatedAaddSplunkforwarder(ctx, sf, "openshift-splunk-forwarder-operator")
+		Expect(apierrors.IsForbidden(err)).To(BeTrue())
+		if err == nil {
+			err = deleteSplunkforwarder(ctx, dedicatedadminsplunkforwarder, operatorNamespace)
+			if err != nil {
+				log.Printf("Failed cleaning up %s CR in %s namespace", dedicatedadminsplunkforwarder, operatorNamespace)
+			}
+		}
+	})
+
+})
+
+// Create test splunkforwarder CR definition
+func makeMinimalSplunkforwarder(kind string, apiversion string, name string) sfv1alpha1.SplunkForwarder {
+	sf := sfv1alpha1.SplunkForwarder{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       kind,
+			APIVersion: apiversion,
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		Spec: sfv1alpha1.SplunkForwarderSpec{
+			SplunkLicenseAccepted: true,
+			UseHeavyForwarder:     false,
+			SplunkInputs: []sfv1alpha1.SplunkForwarderInputs{
+				{
+					Path: "",
+				},
+			},
+		},
+	}
+	return sf
+}
+
+// Create test splunkforwarder CR as a dedicated admin user
+func dedicatedAaddSplunkforwarder(ctx context.Context, SplunkForwarder sfv1alpha1.SplunkForwarder, namespace string) error {
+	restConfig, err := config.GetConfig()
+	obj, err := runtime.DefaultUnstructuredConverter.ToUnstructured(SplunkForwarder.DeepCopy())
+	if err != nil {
+		return err
+	}
+	unstructuredObj := unstructured.Unstructured{obj}
+
+	restConfig.Impersonate = rest.ImpersonationConfig{
+		UserName: "test-user@redhat.com",
+		Groups: []string{
+			"dedicated-admins",
+		},
+	}
+	defer func() {
+		restConfig.Impersonate = rest.ImpersonationConfig{}
+	}()
+	_, err = Dynamic(restConfig).Resource(schema.GroupVersionResource{
+		Group: "splunkforwarder.managed.openshift.io", Version: "v1alpha1", Resource: "splunkforwarders",
+	}).Namespace(namespace).Create(ctx, &unstructuredObj, metav1.CreateOptions{})
```

Comment:
for where we interact with the CRDs, we can instead register the schema and not have to use the subresource client.

something like this where you create the client

```
sfv1alpha1.AddToScheme(client.GetScheme())
```

https://github.com/openshift/ocm-agent-operator/blob/master/osde2e/ocm_agent_operator_tests.go#L51